specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1086 @@
+
+## Subnet Actor (SA)
+The Subnet Actor interface the core functions and basic rules required for an actor to implement the logic for a new subnet. This approach gives users total flexibility to configure the consensus, security assumption, checkpointing strategy, policies, etc. of their new subnet so it fulfills all the needs of their use case.
+
+The Subnet Actor is the public contract accessible by users in the system to determine the kind of child subnet being spawned and controlled by the actor. From the moment the SA for a new subnet is spawned in the parent chain, users looking to participate from the subnet can instantiate their new chain and even start mining on it. However, in order for the subnet to be able to interact with the rest of the hierarchy, it needs to be registered by staking an amount of native tokens over the `CollateralThreshold` in the parent's SCA _(see [Collateral and slashing](#Collateral-and-slashing "Collateral and slashing"))_.
+
+##### Subnet Actor Interface
+A reference implementation for the subnet actor interface exists, but users are permitted to implement custom governing policies and mechanics that better suit their needs (e.g. request a dynamic collateral for validators, add a leaving fee coefficient to penalize validators leaving the subnet before some time frame, require a minimum number of validators, include latency limits, etc.).
+
+> Interface every subnet actors needs to implement. These functions
+> are triggered when a `message` is sent for their corresponding `methodNum`. 
+```go
+type SubnetActor interface{
+    // Initializes the state of the subnet actor and sets
+    // all its initial parameter.
+    // (methodNum = 1)
+    Constructor(ConstructParams)
+    
+    // It expects as `value` the amount of collateral the 
+    // source wants to stake to become part of the subne.
+    // It triggers all check to see determine if the source is elegible
+    // to become part of the subnet, and must propagate an `AddCollateral` message
+    // to the SCA to trigger the state.
+    // 
+    // Join is also responsible to send a `Register` message to the SCA
+    // whenever the `CollateralThreshold` is reached. This function can also
+    // be used to AddCollateral for the subnet in the SCA. Additional
+    // checks for this operation may be included. As a result of the
+    // execution an `AddCollateral` message must be sent to the SCA 
+    // including the added collateral
+    // (methodNum = 2)
+    Join()
+    
+    // Called by participants that want to leave the subnet. It triggers all
+    // the checks to see if the source is eligible to leave the subnet.
+    // This function must propagate a `ReleaseCollateral` message in SCA to 
+    // return their collateral to the actor so a message can be triggered
+    // from the actor to return it back to its owner.
```

Comment:
A bit hard to parse for me
message in SCA ... their collateral ... to the actor ... from the actor ... return it to its owner.